src/targets/gpu/jit/pointwise.cpp

```diff
@@ -46,7 +46,7 @@ struct pointwise_compiler : compiler<pointwise_compiler>
         if(std::any_of(inputs.begin(), inputs.end(), [](const auto& s) { return s.broadcasted(); }))
             return 1;
         else
-            return 4;
+            return 256;
```

Comment:
Well this doesn't do the vectorization, its just a rough calculation to decide if we are vectorizing or not, and adjust the number of elements. In the future, I would like to remove the `auto_vectorize` in the kernel and do the vectorization on the host.

Reply:
For that function, if the number of elements() is not a multiple of 4 or 2, it output is not able to be vectorized (return false). What I mean is, in the future, even it is not a multiple of 2, we can still use the vectorized implementation (like the half2) to implementation some kernels (need additional processing for the remaining elements) since half2 is still faster (with additional processing of the remaining elements)